indico/MaKaC/authentication/baseAuthentication.py

```diff
@@ -287,25 +289,30 @@ def retrieveAvatar(self, rh):
             if personId == '-1':
                 personId = None
             ah = AvatarHolder()
-            av = ah.match({"email": email}, exact=1, onlyActivated=False, searchInAuthenticators=False)
-            if av:
-                av = av[0]
-                # don't allow disabled accounts
-                if av.isDisabled():
-                    return None
-                elif not av.isActivated():
-                    av.activateAccount()
-
-                av.clearAuthenticatorPersonalData()
-                av.setAuthenticatorPersonalData('phone', phone)
-                av.setAuthenticatorPersonalData('fax', fax)
-                av.setAuthenticatorPersonalData('surName', lastname)
-                av.setAuthenticatorPersonalData('firstName', firstname)
-                av.setAuthenticatorPersonalData('affiliation', institute)
-                if personId != None and personId != av.getPersonId():
-                    av.setPersonId(personId)
-            else:
-                avDict = {"email": email,
+            for mail in re.split(";|,", email):
+                av = ah.match({"email": mail}, exact=1, onlyActivated=False, searchInAuthenticators=False)
+                if av:
+                    av = av[0]
+                    # don't allow disabled accounts
+                    if av.isDisabled():
+                        return None
+                    elif not av.isActivated():
+                        av.activateAccount()
+
+                    av.clearAuthenticatorPersonalData()
+                    av.setAuthenticatorPersonalData('phone', phone)
+                    av.setAuthenticatorPersonalData('fax', fax)
+                    av.setAuthenticatorPersonalData('surName', lastname)
+                    av.setAuthenticatorPersonalData('firstName', firstname)
+                    av.setAuthenticatorPersonalData('affiliation', institute)
+                    if personId != None and personId != av.getPersonId():
+                        av.setPersonId(personId)
+
+                    self._postLogin(login, av, True)
+                    return av
+
+            if not av:
+                avDict = {"email": re.split(";|,",email),
```

Comment:
Actually it is not the same with #1638 because if I get correctly you were using only one mail from LDAP to associate the account and the order can create problem. In this case it is not important which mail will be the first because all of them are verified during the login. Whatever is selected as principal mail, when the user try to login again there is a loop through all mails and when one is identified the user is logged in- So in our test (I have multiple mail in my SAML) there are not problem to associate the accounts.

Additionally, the user can replace the preferred mail going to his/her preferences without problem because of the multi mail search.